src/devices/bus/ieee488/hp9133.cpp

```diff
@@ -0,0 +1,878 @@
+}
+
+void hp9133_device::device_reset()
+{
+	memset(m_dma_ram, 0, sizeof(m_dma_ram));
+	m_dma_addr = 0;
+	m_intsel = 0;
+	m_head = 0;
+	m_harddisk_index = false;
+	m_floppy_index_int = false;
+	m_intenable = false;
+	m_dmaenable = false;
+	m_dmaack_switch = false;
+	m_dma_rwn = false;
+	m_gpib_irq = false;
+	m_gpib_drq = false;
+	m_fdc_irq = false;
+	m_fdc_drq = false;
+	m_cpuirq = false;
+	m_cpufirq = false;
+	m_hdc_intrq = false;
+	m_hdc_bdrq = false;
+	m_hdc_bcs = false;
+	m_hdc_wg = false;
+	m_fast = false;
+	m_fdc->set_floppy(m_floppy->get_device());
+}
+
+void hp9133_device::ieee488_eoi(int state)
+{
+	m_gpib->eoi_w(state);
+}
+
+void hp9133_device::ieee488_dav(int state)
+{
+	m_gpib->dav_w(state);
+}
+
+void hp9133_device::ieee488_nrfd(int state)
+{
+	m_gpib->nrfd_w(state);
+}
+
+void hp9133_device::ieee488_ndac(int state)
+{
+	m_gpib->ndac_w(state);
+}
+
+void hp9133_device::ieee488_srq(int state)
+{
+	m_gpib->srq_w(state);
+}
+
+void hp9133_device::ieee488_ifc(int state)
+{
+	m_gpib->ifc_w(state);
+}
+
+void hp9133_device::ieee488_atn(int state)
+{
+	m_gpib->atn_w(state);
+}
+
+void hp9133_device::ieee488_ren(int state)
+{
+	m_gpib->ren_w(state);
+}
+
+void hp9133_device::i8291a_eoi_w(int state)
+{
+	m_bus->eoi_w(this, state);
+}
+
+void hp9133_device::i8291a_dav_w(int state)
+{
+	m_bus->dav_w(this, state);
+}
+
+void hp9133_device::i8291a_nrfd_w(int state)
+{
+	m_bus->nrfd_w(this, state);
+}
+
+void hp9133_device::i8291a_ndac_w(int state)
+{
+	m_bus->ndac_w(this, state);
+}
+
+void hp9133_device::i8291a_srq_w(int state)
+{
+	m_bus->srq_w(this, state);
+}
+
+void hp9133_device::i8291a_int_w(int state)
+{
+	LOG("%s: %d\n", __func__, state);
+	m_gpib_irq = state;
+	update_intsel();
+}
+
+uint8_t hp9133_device::i8291a_dio_r()
+{
+	uint8_t ret = m_bus->dio_r();
+	LOG("%s: %02x\n", __func__, (uint8_t)~ret);
+	return ret;
+}
+
+void hp9133_device::i8291a_dio_w(uint8_t data)
+{
+	m_bus->dio_w(this, data);
+}
+
+void hp9133_device::i8291a_drq_w(int state)
+{
+	LOG("%s: %d\n", __func__, state);
+	m_gpib_drq = state;
+	update_intsel();
+}
+
+TIMER_CALLBACK_MEMBER(hp9133_device::index_timer)
+{
+	attotime next = attotime::from_hz(3600 / 60);
+
+	m_harddisk_index ^= true;
+	if (!m_harddisk_index)
+		next /= 10;
+	else
+		next -= next / 10;
+	m_index_timer->adjust(next);
+}
+
+void hp9133_device::cpu_sync_ack(int state)
+{
+	uint8_t data;
+
+	LOG("%s: %d: enabled %d\n", __func__, state, m_dmaenable);
+	if (!m_dmaenable)
+		return;
+	switch (state) {
+	case ASSERT_LINE:
+		if (m_gpib_drq)
+			m_cpu->set_input_line(M6809_IRQ_LINE, ASSERT_LINE);
+		break;
+	case CLEAR_LINE:
+		if (!m_dma_rwn) {
+			data = dma_ram_r(0);
+			m_gpib->dout_w(data);
+		} else {
+			data = m_gpib->din_r();
+			dma_ram_w(data);
+		}
+		m_cpu->set_input_line(M6809_IRQ_LINE, CLEAR_LINE);
+		LOG("%s: sync data: %02x\n", __func__, data);
+		if (m_dma_addr > 256) {
+			LOG("DMA: all bytes transferred\n");
+			m_dmaenable = false;
+		}
+		break;
+	}
```

Comment:
You’ll only see 1/0 values in something attached to a `devcb_write_line` – you can just use `if (state) { ...`.

Reply:
Ok, i found the switch easier to read than a large if/else, but doesn't really matter.